### ISSUE TITLE
Support a tombstone message for a delete operation

### ISSUE THREAD

**Issue by NathanNam** [microsoft/kafka-connect-cosmosdb#411]
**## Problem Statement**
Debezium CDC Source Connector and other source connectors allow having a tombstone record (a Kafka record key exists but a Kafka record value is empty). We can follow a similar approach we do for Cassandra Sink connector.

**## Proposed Solution**

**## Alternative Solutions**

**## Additional context**

## Next Steps

- [ ] Team consensus to proceed
- [ ] Schedule Design Session
- [ ] Complete Design Review

**Comment by NathanNam** [comment]
@ryancrawcour No. Users can have a tombstone record (A null payload is a payload with 0 bytes.) and CosmosDB Sink Connector can read this tombstone record and delete the relevant record in Cosmos DB.

**Comment by ryancrawcour** [comment]
> Or, are you trying to initiate a delete operation in the database by placing a "tombstone" message in to a Kafka topic?

thanks @NathanNam so it is this scenario then ☝

**Comment by NathanNam** [comment]
Pk should come from a Kafka record key. I think we have discussed this part a few months ago. As far as I recall, Cosmos DB needs a primary key and a partition key to locate the right record.

**Comment by ryancrawcour** [comment]
so doc id (or primary key) **and** partition key should both be set for Cosmos DB to do delete efficiently.  
so you want them to **both** come from the Kafka key (and have 0 bytes in the body / value of the message)

Can you share a sample of a tombstone message that you would use that contains the id and the partition-key?

**Comment by javierholguera** [comment]
Picking up on this. Tombstone messages aleways have a "null" value in the Kafka record value. The key can be whatever needs to be. I imagine in this case, since it requires two values in the key (ID and partition key), the key would be something like an Avro structure or JSON payload that contains both and the connector can be configured to use the appropriate Key (de)Serializer and paths to both properties in them.

Obviously, that would force any connector that wants to use hard delete functionality to always use a Key format that supports multiple fields (again, something like JSON or Avro comes to mind). This is not that strange in the Kafka ecosystem anyway, the Debezium connector mentioned earlier does that (for similar reasons).
